src/algebra/Mon/basic.lean

```diff
@@ -33,6 +33,14 @@ abbreviation forget : Mon.{u} ⥤ Type u := forget
 
 instance hom_is_monoid_hom {R S : Mon} (f : R ⟶ S) : is_monoid_hom (f : R → S) := f.2
 
+/-- Morphisms in `Mon` are defined using `subtype is_monoid_hom`,
+so we provide a canonical bijection with `R →* S`. -/
+def hom_equiv_monoid_hom (R S : Mon) : (R ⟶ S) ≃ (R →* S) :=
+{ to_fun := λ f, ⟨f, f.2.2, f.2.1.1⟩,
```

Comment:
I'm not very fond of things like `2.1.1`. It is very non-descriptive. I guess this could also be written as something like `f.property.map_mul` or something like that?